### PR TITLE
Small fixes in session, h2quic

### PR DESCRIPTION
**Pull request by jbenoist**
- populateServerConfig() doesn't honor the keepalive setting of the caller
- h2quic seems to pass as TCP connection instead of TLS to the http2 stack in the non-quic case

### CODE REVIEW

**Comment by lucas-clemente** [comment]
LGTM, but we need you to sign the CLA before merging. Let me know if you have any questions.

**Comment by jbenoist** [comment]
> On Sep 22, 2017, at 3:40 AM, Lucas Clemente <notifications@github.com> wrote:
> 
> LGTM, but we need you to sign the CLA before merging. Let me know if you have any questions.

I signed it actually, but it turns out that the email address of my google account is different (e.g: langton@benoist.name).
If this is problematic, my employer has a corporate CLA that might be able to use. Just let me know.

—
Julien Benoist

**Comment by jbenoist** [comment]
the address should match now, I just updated it.

File: server.go [hunk 1]
```diff
@@ -149,6 +149,7 @@ func populateServerConfig(config *Config) *Config {
 		HandshakeTimeout:                      handshakeTimeout,
 		IdleTimeout:                           idleTimeout,
 		AcceptSTK:                             vsa,
+		KeepAlive:                             config.KeepAlive,
```
Comment (marten-seemann) [inline server.go]: Can you please add a test for this value in the two tests here: https://github.com/lucas-clemente/quic-go/blob/master/server_test.go#L344-L373.

**Comment by jbenoist** [comment]
If the second function you mentioned happens to be the one that checks whether the defaults are correctly used when options are not set, there isn't one for keepalive since type default suffices.

**Comment by marten-seemann** [comment]
@jbenoist: Don't worry about the merge conflict. I'll take care of that.

**Comment by marten-seemann** [comment]
Closing in favor of #842.

**Pull request closed without merging**